src/NWAClient.ts

```diff
@@ -0,0 +1,236 @@
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { generateSecretKey, getPublicKey, Relay } from "nostr-tools";
+import {
+  BudgetRenewalPeriod,
+  Nip47Method,
+  Nip47NetworkError,
+  Nip47NotificationType,
+  NWCClient,
+} from "./NWCClient";
+
+export type NWAOptions = {
+  relayUrl: string;
+  appPubkey: string;
+
+  name?: string;
+  icon?: string;
+  requestMethods: Nip47Method[];
```

Comment:
(minor) this required field should be shifted to the top 😬

Reply:
oops, fixed :pray: